src/utils/containers.ts

```diff
@@ -35,28 +109,28 @@ export class SelfClearingSet<T> {
     }
 }
 
-export class SelfClearingMap<K, V> {
+export class SelfClearingMap<K, V> extends SelfClearingContainer {
     contents = new Map<K, [number, V]>();
     duration: number;
-    interval: NodeJS.Timeout;
     on_remove?: (key: K, value: V) => void;
     constructor(duration: number, interval?: number, on_remove?: (key: K, value: V) => void) {
+        super(interval ?? duration);
         this.duration = duration;
-        this.interval = set_interval(this.sweep.bind(this), interval ?? this.duration);
         this.on_remove = on_remove;
```

Comment:
The reason on_remove was added was so I could manually RAII-style cleanup things in the case of a `SelfClearingMap<string, SelfClearingMap<string, SelfClearingSet<string>>>`. It's probably no longer needed, however it doesn't hurt to have the functionality.

Reply:
I thought the same, that while it's outlived its current usefulness, it might be a good idea to keep it in case another process needs more cleanup